### ISSUE TITLE
Remove starting_items redundancy in plando/spoiler

### ISSUE THREAD

**Issue by fenhl** [TestRunnerSRL/OoT-Randomizer#1349]
# Reasoning

* There are currently two ways to specify starting items in a plando: The top-level entry `starting_items`, as well as the `starting_items` inside `settings`. This is somewhat redundant and confusing.
* Having a list of all effective starting items, including Link's Pocket and Song from Impa (if Skip Child Zelda is on), would not only be useful for readers of spoiler logs, but also in the rest of the code, e.g.:
    * Simplifying the Song from Impa handling in #1274
    * Fixing the bug where a Triforce piece on free Song from Impa overwrites plando'd starting pieces instead of being added
    * Making it easier to handle Link's Pocket if and when medallion shuffle is added

# Steps

- [ ] Allow the `starting_items` setting to be plando'd with the syntax currently used by the top-level `starting_items` entry
    * This will be in addition to the current syntax for the setting, which will continue to be used by the GUI for now
- [ ] Remove the top-level entry, replacing it with an informative (i.e. ignored in plando) `:effective_starting_items` entry
- [ ] Modify the GUI to add any features that are currently exclusive to plando

**Comment by r0bd0g** [comment]
How would it know whether the starting items defined there would be from impa or not? Like if I say I want something in starting items, and the a copy of that item goes on skipped impa, expected behaviour I think is that you'd start with 2 copies.

**Comment by fenhl** [comment]
Right, and that behavior is currently broken (at least with Triforce pieces). I think patching starting items in the rom from the effective starting items list should fix that issue.

**Comment by r0bd0g** [comment]
On the other hand let's say I plando a starting song. Most people might expect that to be the song from impa. But what could happen instead is songs as items is enabled and one of the song checks is replaced by a randomized item, which may or may not go onto the impa check.

**Comment by r0bd0g** [comment]
Maybe a more robust way of marking checks as 'skipped' could be helpful. And then you'd have a non-plandoable section in the spoiler for the items obtained from skipped checks. I could see reimplementing link's pocket even as a skipped check in tot, idk. I'm not sure how combining these two types of starting items is going to help?

**Comment by fenhl** [comment]
I think we can reasonably expect plando users to know the difference between the song from Impa and true starting items. Changing plando starting items to use that check sounds like it would require substantial changes to the code.

**Comment by fenhl** [comment]
A skipped checks system sounds like a reasonable future change that could build upon this. For examples of how combining them could help, see the second bullet point in the opening comment above.

**Comment by flagrama** [comment]
@r0bd0g 
> I'm not sure how combining these two types of starting items is going to help?

Top level `starting_items` is effectively a relic of pre-GUI-Starting-Items used exclusively for plando. It doesn't really make sense to keep it aside from the GUI not yet being able to do _everything_ that the top level `starting_items` can. It would make much more sense to reuse it as an informational "these are all the items you have after creating the save file yet not playing it yet." That information is useful both for someone viewing a spoiler log, and for having a specific place to check while writing code.

**Comment by r0bd0g** [comment]
I just don't see how you can combine the starting items with the items from skipped checks. You need some way to specify which is the item from the skipped check. So I think starting items and items from skipped checks probably ought to stay separate.

**Comment by fenhl** [comment]
The spoiler log field that includes skipped checks will start with `:`, i.e. be ignored for plando, like `:settings_string` or `:playthrough`.

**Comment by Roman971** [comment]
I think I agree with robdog that including skipped checks in starting items is confusing, even if it's in a section that is ignored when using the spoiler log as plando. The main issue is you can't really tell whether the starting item is from a skipped check or not, and you end up with items that are both in locations and in effective starting items which would be very confusing to a new user.

Also, like flagrama mentioned, the top-level `starting_items` currently supports a lot more than the settings `starting_items`, so I don't see how we can get rid of it. Tho, if I understand correctly, the idea isn't to get rid of it but to somehow allow users to use the old syntax in the settings `starting_items`, but I honestly don't see how that's any less confusing than the current situation, or how users should expect to be able to do this.

I don't really have a better idea right now, but I feel like it could make more sense to keep only the top-level `starting_items` since it can do a lot more, and change the settings starting items to use the top-level one somehow to remove the redundancy.

**Comment by fenhl** [comment]
> I think I agree with robdog that including skipped checks in starting items is confusing, even if it's in a section that is ignored when using the spoiler log as plando. The main issue is you can't really tell whether the starting item is from a skipped check or not, and you end up with items that are both in locations and in effective starting items which would be very confusing to a new user.

The idea behind the name “effective starting items” is that items from skipped checks are effectively starting items (though not from the setting) so these are included here.

> Also, like flagrama mentioned, the top-level `starting_items` currently supports a lot more than the settings `starting_items`, so I don't see how we can get rid of it. Tho, if I understand correctly, the idea isn't to get rid of it but to somehow allow users to use the old syntax in the settings `starting_items`, but I honestly don't see how that's any less confusing than the current situation, or how users should expect to be able to do this.

My PR addresses this by allowing the `starting_items` setting to be specified with the syntax previously used for the top-level entry. The old setting syntax is kept as well since that's what the GUI uses. I could see the GUI eventually moving to the more powerful syntax and the old setting syntax being retired.

> I don't really have a better idea right now, but I feel like it could make more sense to keep only the top-level `starting_items` since it can do a lot more, and change the settings starting items to use the top-level one somehow to remove the redundancy.

The issues I see with this approach is that it wouldn't address the bugs fixed by introducing `:effective_starting_items`, and that it would introduce a setting accessible via the GUI that isn't part of the `settings` object.

**Comment by Roman971** [comment]
> The idea behind the name “effective starting items” is that items from skipped checks are effectively starting items (though not from the setting), since the default save file does literally start with them, so these are included here.

Sure, but in that section you would also have all the actual starting items, right? So the user couldn't tell which ones are from skipped locations and which ones are from settings/plando, and this is where the confusion lies for me.
In order to read the spoiler log properly, you would basically have to know exactly which settings cause which locations to be skipped, and you also need to know that the items from skipped locations are treated as starting items, even tho it wasn't the case so far (as far as the user is concerned).

**Comment by Zannick** [comment]
Maybe the compromise here is to add a separate ":free_locations" or ":skipped_locations" that includes only the skipped locations that provide items at the start? eg. maybe Link's Pocket would always be there.

**Comment by fenhl** [comment]
I would like to point out that there is a 3rd source for effective starting items: ammo for items on skipped locations and old-syntax starting items. Maybe having a separate `:skipped_locations` list but still listing effective starting items as a unit would work?

**Comment by fenhl** [comment]
For the record, this was discussed again on Discord and I ended up removing `:effective_starting_items` and replacing it with a `:skipped_locations` mapping.

**Comment by fenhl** [comment]
For part 2, the GUI would essentially have to be changed to use a subset of the item list instead of the current hardcoded list and to allow a way to specify a count for each selected starting item.
